packages/splitter/src/useSplitter.ts

```diff
@@ -25,18 +31,31 @@ interface IEventTargetLike {
 
 interface IBodyLike extends IEventTargetLike {
   clientWidth: number;
+  clientHeight: number;
 }
 
 interface IDocumentLike extends IEventTargetLike {
   body: IBodyLike;
 }
+
+export interface IWindowLike {
+  document: IDocumentLike;
+  scrollX: number;
+  scrollY: number;
+}
+
+export interface ISplitterGetterReturnProps extends HTMLProps<any> {
+  'data-garden-container-id': string;
+  'data-garden-container-version': string;
+}
+
 export interface IUseSplitterProps extends Omit<HTMLProps<any>, 'onChange'> {
+  /** Prefixes IDs for the separator */
+  idPrefix?: string;
   /** An aria-label for the separator */
   ariaLabel?: string;
-  /** An id of the element the separator controls */
-  primaryPaneId: string;
-  /** A document environment to attach events to */
-  environment?: Document | IDocumentLike;
+  /** A browser window environment to attach events to */
+  environment: Window | IWindowLike;
```

Comment:
This feels right. Apologies for past misdirection.

Reply:
No worries. It was necessary to mold the architectural decisions for this component. Im sure we are richer for those conversations.